pkg/client/client.go

```diff
@@ -325,7 +329,24 @@ func New(c *restclient.Config) (*Client, error) {
 		return nil, err
 	}
 
-	return &Client{client}, nil
+	return &Client{RESTClient: client, config: &config}, nil
+}
+
+// ForGroupVersion returns a copy of the client but without any openshift legacy
+// defaulting.
+// NOTE: This client should be only used in transition period when the legacy API is
+// locked for new types, but the clientsets are not provided.
+func (c *Client) ForGroupVersion(gv *unversioned.GroupVersion) *Client {
```

Comment:
`ForGroupVersion(kapi.SchemeGroupVersion.GroupVersion())` will not be correct. But as this is only for transition we shouldn't care maybe.